### PR TITLE
Verbs: Enhanced ODP functionality 

### PR DESCRIPTION
**Pull request by yishaih**
This series enhanced the ODP functionality in the user space area as follows:
- It reports ODP related capabilities on XRC and SRQ.
- It handles some page fault flow in the mlx5 driver .

The matching kernel part was already merged into for-next.

### CODE REVIEW

File: libibverbs/cmd.c [hunk 1]
```diff
@@ -265,6 +265,15 @@ int ibv_cmd_query_device_ex(struct ibv_context *context,
 		}
 	}
 
+	if (attr_size >= offsetof(struct ibv_device_attr_ex, xrc_odp_caps) +
```
Comment (jgunthorpe) [inline libibverbs/cmd.c]: All these attr_size checks are obsolete and should be completely deleted. Now that verbs and the providers share the same code it is not possible for attr to not have xrc_odp_caps.

File: providers/mlx5/srq.c [hunk 2]
```diff
@@ -179,28 +293,38 @@ int mlx5_alloc_srq_buf(struct ibv_context *context, struct mlx5_srq *srq)
...
+		srq->waitq_head = -1;
+		srq->waitq_tail = -1;
 	}
 
+	srq->wrid = malloc(srq->max * sizeof(*srq->wrid));
+	if (!srq->wrid) {
+		mlx5_free_buf(&srq->buf);
+		return -1;
+	}
 	memset(srq->buf.buf, 0, buf_size);
```
Comment (jgunthorpe) [inline providers/mlx5/srq.c]: This memset is now in a goofy place

File: libibverbs/cmd.c (hunk 1 above)
Comment (yishaih) [reply libibverbs/cmd.c]: 'attr_size' is the application's input size which comes from the static inline function
ibv_query_device_ex(),  applications that were complied with legacy libibverbs may not have this field in place, that's why this check is still needed here as done also above.

File: providers/mlx5/srq.c (hunk 2 above)
Comment (yishaih) [reply providers/mlx5/srq.c]: Correct, will move to the original place.

**Pull request merged**